code/game/objects/structures/aliens.dm

```diff
@@ -176,14 +176,19 @@
 /obj/structure/alien/weeds/node
 	name = "glowing resin"
 	desc = "Blue bioluminescence shines from beneath the surface."
-	icon_state = "weednode-0"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "weednode"
 	base_icon_state = "weednode"
 	light_color = LIGHT_COLOR_BLUE
 	light_power = 0.5
 	var/lon_range = 4
 	var/node_range = NODERANGE
+	pixel_x = 0
+	pixel_y = 0
```

Comment:
Use `MAP_SWITCH` instead of `Initialize`. Also afaik this will break overlays? You should use transform like volcanic walls do, or at least basepixels if there's no overlays ever.

Reply:
It seems like the Initialize logic acts as a wierd stopgap to prevent inherited initialization logic from overriding the icon when randomizing appearance using weeds1.dmi weeds2.dmi and weeds3.dmi; I'll see what I can do to improve this code while I'm here.